src/encoder/plugins/FdkaacEncoderPlugin.cxx

```diff
@@ -0,0 +1,263 @@
+
+	~FdkaacEncoder() override;
+
+	/* virtual methods from class Encoder */
+	void Write(const void *data, size_t length) override;
+	size_t Read(void *dest, size_t length) override;
+};
+
+class PreparedFdkaacEncoder final : public PreparedEncoder {
+	HANDLE_AACENCODER *AacEncoder;
+	AUDIO_OBJECT_TYPE aot;
+	int bitrate;
+	int quality;
+	bool aacenc_afterburner;
+
+public:
+	PreparedFdkaacEncoder(const ConfigBlock &block);
+
+	/* virtual methods from class PreparedEncoder */
+	Encoder *Open(AudioFormat &audio_format) override;
+
+	const char *GetMimeType() const override {
+		fprintf(stderr, "GetMimeType\n");
+		return "audio/aac";
+	}
+};
+
+PreparedFdkaacEncoder::PreparedFdkaacEncoder(const ConfigBlock &block)
+{
+	const char *value;
+
+	value = block.GetBlockValue("aot", "lc");
+	if (value != nullptr) {
+		if(!strcmp(value, "lc")) {
+			aot=AOT_AAC_LC;
+		}else if(!strcmp(value, "he")) {
+			aot=AOT_SBR;
+		}else if(!strcmp(value, "hev2")) {
+			aot=AOT_PS;
+		}else if(!strcmp(value, "ld")) {
+			aot=AOT_ER_AAC_LD;
+		}else if(!strcmp(value, "eld")) {
+			aot=AOT_ER_AAC_ELD;
+		}else {
+			aot=AOT_AAC_LC;
+		}
+	}
+	aacenc_afterburner = block.GetBlockValue("aacenc_afterburner", true);
+	quality = block.GetBlockValue("quality", 0);
+	bitrate = block.GetBlockValue("bitrate", 128)*1000;
+}
+
+static PreparedEncoder *
+fdkaac_encoder_init(const ConfigBlock &block)
+{
+	return new PreparedFdkaacEncoder(block);
+}
+
+static void
+fdkaac_encoder_setup(HANDLE_AACENCODER *AacEncoder, AUDIO_OBJECT_TYPE aot, int bitrate,
+int quality, bool aacenc_afterburner, const AudioFormat &audio_format)
+{
+	if(aacEncoder_SetParam(*AacEncoder, AACENC_AOT, aot) != AACENC_OK) {
+		throw std::runtime_error("error setting fdkaac AOT");
+	}
+	if(aacEncoder_SetParam(*AacEncoder, AACENC_BITRATE, bitrate) != AACENC_OK) {
+		throw std::runtime_error("error setting fdkaac bitrate");
+	}
+	if(aacEncoder_SetParam(*AacEncoder, AACENC_BITRATEMODE, quality) != AACENC_OK) {
+		throw std::runtime_error("error setting fdkaac bitrate mode");
+	}
+	if(aacEncoder_SetParam(*AacEncoder, AACENC_SAMPLERATE, audio_format.sample_rate) != AACENC_OK) {
+		throw std::runtime_error("error setting fdkaac samplerate");
+	}
+	if(aacEncoder_SetParam(*AacEncoder, AACENC_CHANNELMODE, audio_format.channels) != AACENC_OK) {
+		throw std::runtime_error("error setting fdkaac channels");
+	}
+	if(aacEncoder_SetParam(*AacEncoder, AACENC_AFTERBURNER, aacenc_afterburner) != AACENC_OK) {
+		throw std::runtime_error("error setting fdkaac afterburner");
+	}
+	if (aacEncEncode(*AacEncoder, NULL, NULL, NULL, NULL) != AACENC_OK) {
+		throw std::runtime_error("Unable to initialize the encoder\n");
+	}
+}
+
+Encoder *
+PreparedFdkaacEncoder::Open(AudioFormat &audio_format)
+{
+	audio_format.format = SampleFormat::S16;
+	audio_format.channels = 2;
+	AACENC_InfoStruct *info = (AACENC_InfoStruct *)calloc(sizeof(AACENC_InfoStruct), 1);
+	AacEncoder = (HANDLE_AACENCODER*)calloc(sizeof(HANDLE_AACENCODER), 1);
```

Comment:
Does the library really require zero-initialization? Or what is the justification for using this unsafe plain C function?